### DISCUSSION TITLE
Importing tailwindcss/plugin results in a type error when using resolveConfig?

### DISCUSSION THREAD

**Discussion by samesfahani-tuplehealth** [tailwindlabs/tailwindcss#10864] [Help] [answered]
https://codesandbox.io/s/great-boyd-5hiwdk?file=/tailwind.config.js

I'm unsure if this is a bug, but if you look at the sandbox, just having `const plugin = require('tailwindcss/plugin');` at the top of the tailwind config file is seemingly changing some typing when trying to include it elsewhere to resolveConfig.

**Comment by wongjn** [comment, marked as answer]
Its because the `/** @type … */` annotation becomes attached to the `const plugin = …` statement rather than the `module.exports = {` statement. Changing it like so should resolve the type error:

```diff
+const plugin = require("tailwindcss/plugin");
+
 /** @type {import('tailwindcss').Config} */
-
-// Including this file causes TS error next door in tailwindConfig.ts?
-const plugin = require("tailwindcss/plugin");
-
 module.exports = {
   content: ["./src/**/*.{js,jsx,ts,tsx}"],
```

**Reply by samesfahani-tuplehealth** [reply]
That did the trick, thanks! Also needed `allowJS: true` in tsconfig `compilerOptions`.
